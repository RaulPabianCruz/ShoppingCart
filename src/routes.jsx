import Container from './components/Container/Container';
import Home from './components/Home/Home';
import Shop from './components/Shop/Shop';
import Cart from './components/Cart/Cart';
import ErrorPage from './ErrorPage';

const routes = [
  {
    path: '/',
    element: <Container />,
    errorElement: <ErrorPage />,
    children: [
      { index: true, element: <Home /> },
      {
        path: 'shop',
        element: <Shop />,
      },
      {
        path: 'cart',
        element: <Cart />,
      },
    ],
  },
];

export default routes;
