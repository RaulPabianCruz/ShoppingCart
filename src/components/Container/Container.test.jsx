import React from 'react';
import { render, screen } from '@testing-library/react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import routes from '../../routes';

describe('Container Component', () => {
  describe('Container content', () => {
    it('matches snapshot', () => {
      const router = createMemoryRouter(routes, { intialEntries: ['/'] });
      const { container } = render(<RouterProvider router={router} />);

      expect(container).toMatchSnapshot();
    });

    it('renders cart item counter', () => {
      const router = createMemoryRouter(routes, { intialEntries: ['/'] });
      render(<RouterProvider router={router} />);

      const itemCount = screen.getByTestId('num-of-items');

      expect(itemCount.textContent).toBe('0');
    });
  });

  describe('Container links', () => {
    it('renders Home link', () => {
      const router = createMemoryRouter(routes, { initialEntries: ['/'] });
      render(<RouterProvider router={router} />);

      const HomeLink = screen.getByRole('link', { name: 'Home' });

      expect(HomeLink).toBeInTheDocument();
      expect(HomeLink).toHaveAttribute('href', '/');
    });

    it('renders Shop link', () => {
      const router = createMemoryRouter(routes, { intialEntries: ['/'] });
      render(<RouterProvider router={router} />);

      const ShopLink = screen.getByRole('link', { name: 'Shop' });

      expect(ShopLink).toBeInTheDocument();
      expect(ShopLink).toHaveAttribute('href', '/shop');
    });

    it('renders Cart link', () => {
      const router = createMemoryRouter(routes, { intialEntries: ['/'] });
      render(<RouterProvider router={router} />);

      const CartLink = screen.getByRole('link', { name: 'Cart' });

      expect(CartLink).toBeInTheDocument();
      expect(CartLink).toHaveAttribute('href', '/cart');
    });
  });
});
