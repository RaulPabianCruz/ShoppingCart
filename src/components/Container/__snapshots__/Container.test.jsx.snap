// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container Content > snapshot > matches snapshot 1`] = `
<div>
  <div>
    <header
      class="_header_523845"
    >
      <img
        alt="The e-Mart Logo"
        class="_logo_523845 _headerItem_523845"
        src="/src/assets/eMartLogo.webp"
      />
      <h1
        class="_titleName_523845 _headerItem_523845"
      >
        The e-Mart!
      </h1>
      <a
        class="_links_523845 _headerItem_523845"
        href="/"
      >
        Home
      </a>
      <a
        class="_links_523845 _headerItem_523845"
        href="/shop"
      >
        Shop
      </a>
      <p
        class="_itemCount_523845 _headerItem_523845"
        data-testid="num-of-items"
      >
        0
      </p>
      <a
        class="_links_523845 _headerItem_523845"
        data-testid="cart-link"
        href="/cart"
      >
        <img
          alt="Cart"
          class="_icon_523845 _headerItem_523845"
          src="/src/assets/cart.svg"
        />
      </a>
    </header>
    <div>
      <h1>
        Placeholder Home
      </h1>
      <a
        href="/"
      >
        yo waddup bruh
      </a>
    </div>
  </div>
</div>
`;
